x-pack/legacy/plugins/actions/server/lib/action_executor.ts

```diff
@@ -105,9 +127,18 @@ export class ActionExecutor {
         config: validatedConfig,
         secrets: validatedSecrets,
       });
+      loggedEvent.message = `action executed successfully: ${actionLabel}`;
     } catch (err) {
-      logger.warn(`action executed unsuccessfully: ${actionLabel} - ${err.message}`);
-      throw err;
+      error = err;
+      loggedEvent.message = `action executed unsuccessfully: ${actionLabel} - ${err.message}`;
+      logger.warn(loggedEvent.message);
+    }
+
+    eventLogger.stopTiming(loggedEvent);
+    eventLogger.logEvent(loggedEvent);
+
+    if (error != null) {
```

Comment:
This is the only real-life example of logging events right now, there will be more :-)

So, we should figure out a pattern that everyone's happy with, since we'll see more of this, and this usage is likely to be copied.

I'm not really happy with the current flow, nor Gidi's suggestion (duplicate the calls), so will noodle on this a bit more.  Perhaps there's some kind of helper we can build to make this easier.

Reply:
I've created a logger method that will wrap a function execution to handle the timing/logging.  That way we have this yucky code (but went with your style of "two places" for the event logging - error or not).  No doubt we'll tweak this a bit more over time, but should make it easier to apply event logging at existing call sites.

Reply:
Turns out the code I had in at the time of the review was not quite right anyway.  An action could throw an exception, but it can also return a result which indicates an error occurred, with more data provided.  So my "wrapper" for a function execution wasn't useful here.  But ... I think I cleaned this up quite a bit in the current set of commits.

Reply:
Oh woops, this actually needed additional work, as the reviewed code didn't handle the case for errors returned from the action outside of an exception (via `status = 'error'`).  So the wrapper wasn't useful (yet), so I deleted it, maybe we'll find a use later, but ... not sure - it was easy to write.

So that code needed a re-write, hopefully it's a lot clearer, and has no `null` checks :-)